### PR TITLE
fix: update `pushState` function to reflect the new url in the `page` store

### PR DESCRIPTION
**Pull request by SunPodder**
fixes https://github.com/sveltejs/kit/issues/11492

The `pushState` function was not updating the built-in `page` store properties (`$pages.url` and `$pages.route`) even though the actual URL in the browser was changed. This causes changes to not reflected on the UI if a component depends on any of those `page` store properties.

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

**Comment by eltigerchino** [comment]
Thank for submitting this. Do you mind adding a test for it too?

**Comment by dummdidumm** [comment]
@Rich-Harris was it intended to be that way?
I'd like to clarify this first before we merge this - this might be a feature, not a bug.

**Comment by SunPodder** [comment]
> Thank for submitting this. Do you mind adding a test for it too?

Also updated the `replaceState` function. Added tests for both.

File: .changeset/sharp-tigers-pay.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@sveltejs/kit": patch
+---
+
+fix: Update `pushState` function to reflect the new url in the `page` store.
```
Comment (benmccann) [inline .changeset/sharp-tigers-pay.md:5]: please remove one of the duplicate changesets

File: .changeset/rich-dryers-design.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@sveltejs/kit": patch
+---
+
+fix: Update `replaceState` function to reflect the new url in the `page` store
```
Comment (benmccann) [inline .changeset/rich-dryers-design.md:5]: assuming I understood correctly, we could write this a bit more succinctly as:
```suggestion
fix: update `page` store when `replaceState` is called
```

**Comment by Rich-Harris** [comment]
`$page.url` and `$page.route` should not be updated when using `pushState`, because you're not actually changing the route

**Pull request closed without merging**